modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -426,4 +427,18 @@ public static String getSourceName(Class<? extends Source> sourceType)
                              p.getModelProperty(MetadataModelProperty.class).get().isMetadataKeyParam())
                 .findFirst();
     }
+
+    /**
+     * Looks for the annotation in the given class. If the annotation is not found, it keeps looking recursively
+     * for it in the superClass until it finds it or there is no superClass to analyze.
+     */
+    public static <T extends Annotation> T getAnnotationFromClassHierarchy(Class<?> annotatedClass, Class<T> annotationClass){
+        T annotation = annotatedClass.getAnnotation(annotationClass);
+        if (annotation != null){
+            return annotation;
+        }
+
+        return annotatedClass.getSuperclass() != null && !annotatedClass.getSuperclass().isAssignableFrom(Object.class) ?
```

Comment:
all superclasses are assignable from Object.. just check that the superclass is not equals to Object